src/components/NameForm.tsx

```diff
@@ -0,0 +1,40 @@
+import React, {ChangeEventHandler, SyntheticEvent} from "react";
+import ReactDOM from "react-dom";
+
+type Props = { callback: (arg: number) => void };
+type State = { value: number };
+
+class NameForm extends React.Component<Props, State> {
+    handleChange: ChangeEventHandler<HTMLInputElement>;
+    //handleChange: {event: SyntheticEvent<Element, Event>};
+    handleSubmit: () => void;
```

Comment:
make this handleSubmit typed as *EventHandler.
Probably SubmitEventHandler or something like that.